cmd/erasure-object.go

```diff
@@ -1088,37 +1088,54 @@ func (er erasureObjects) DeleteObjects(ctx context.Context, bucket string, objec
 		writeQuorums[i] = getWriteQuorum(len(storageDisks))
 	}
 
-	versions := make([]FileInfo, len(objects))
+	versionsMap := make(map[string]FileInfoVersions)
 	for i := range objects {
+		vr := FileInfo{
+			Name:             objects[i].ObjectName,
+			VersionID:        objects[i].VersionID,
+			ReplicationState: objects[i].ReplicationState(),
+			Idx:              i,
+		}
+		vr.SetTierFreeVersionID(mustGetUUID())
 		if objects[i].VersionID == "" {
-			modTime := opts.MTime
-			if opts.MTime.IsZero() {
-				modTime = UTCNow()
-			}
-			uuid := opts.VersionID
-			if uuid == "" {
-				uuid = mustGetUUID()
-			}
 			if opts.Versioned || opts.VersionSuspended {
-				versions[i] = FileInfo{
-					Name:             objects[i].ObjectName,
-					ModTime:          modTime,
-					Deleted:          true, // delete marker
-					ReplicationState: objects[i].ReplicationState(),
-				}
-				versions[i].SetTierFreeVersionID(mustGetUUID())
+				vr.ModTime = UTCNow()
+				vr.Deleted = true
 				if opts.Versioned {
-					versions[i].VersionID = uuid
+					vr.VersionID = mustGetUUID()
```

Comment:
I don't really understand what is going on here. Could you add an explanation?

Reply:
yes added.